src/syscall/syscall_libc.go

```diff
@@ -26,15 +31,29 @@ func Write(fd int, p []byte) (n int, err error) {
 }
 
 func Read(fd int, p []byte) (n int, err error) {
-	return 0, ENOSYS // TODO
+	buf, count := splitSlice(p)
+	n = libc_read(int32(fd), buf, uint(count))
+	if n < 0 {
+		err = getErrno()
+	}
+	return
 }
 
 func Seek(fd int, offset int64, whence int) (off int64, err error) {
 	return 0, ENOSYS // TODO
 }
 
 func Open(path string, mode int, perm uint32) (fd int, err error) {
-	return 0, ENOSYS // TODO
+	buf, count := splitSlice([]byte(path))
+	cstr := libc_strndup(buf, count)
+	defer libc_free(cstr)
```

Comment:
You can avoid a heap allocation here by avoiding the string to []byte conversion. You can create a new `splitString` function and pass the resulting pointer directly to `libc_strndup`.